stable_baselines/common/policies.py

```diff
@@ -111,17 +111,24 @@ def __init__(self, sess, ob_space, ac_space, n_env, n_steps, n_batch, reuse=Fals
                  obs_phs=None, add_action_ph=False):
         self.n_env = n_env
         self.n_steps = n_steps
-        self.n_batch = n_batch
         with tf.variable_scope("input", reuse=False):
             if obs_phs is None:
                 self._obs_ph, self._processed_obs = observation_input(ob_space, n_batch, scale=scale)
             else:
                 self._obs_ph, self._processed_obs = obs_phs
-
             self._action_ph = None
             if add_action_ph:
                 self._action_ph = tf.placeholder(dtype=ac_space.dtype, shape=(n_batch,) + ac_space.shape,
                                                  name="action_ph")
+            if isinstance(ac_space, spaces.MultiDiscrete):
+                self._action_mask_ph = tf.placeholder(dtype=tf.float32, shape=(n_batch, ac_space.nvec[0],
+                                                                               ac_space.nvec[1]), name="action_mask_ph")
+            elif isinstance(ac_space, spaces.Discrete) or isinstance(ac_space, spaces.MultiBinary):
+                self._action_mask_ph = tf.placeholder(dtype=tf.float32, shape=(n_batch, ac_space.n),
+                                                      name="action_mask_ph")
+            elif isinstance(ac_space, spaces.Box):
+                self._action_mask_ph = tf.placeholder(dtype=tf.float32, shape=(n_batch, ac_space.shape[0]),
+                                                      name="action_mask_ph")
```

Comment:
Pretty self documenting. Different action spaces call for different dimension ed action masks.